### PR TITLE
Service provider docs

### PR DESCRIPTION
**Pull request by deric**
Originally I wanted to add just the documentation. Then I came up with few more modifications. Service provider is not set, unless needed. Such approach is also recommended by Puppet docs:

> You will seldom need to specify this — Puppet will usually discover the appropriate provider for your platform.

If you really want to have some default value, we should add something like `params.pp` with appropriate service provider for each supported distribution.

### CODE REVIEW

File: spec/classes/aerospike_spec.rb [hunk 1]
```diff
@@ -516,8 +518,7 @@
             .with_ensure('stopped')\
             .with_enable(true)\
             .with_hasrestart(true)\
-            .with_hasstatus(true)\
-            .with_provider('init')
```
Comment (aerostitch) [inline spec/classes/aerospike_spec.rb]: Why removing the the provider check here?

File: .gitignore [hunk 2]
```diff
@@ -5,3 +5,5 @@ spec/fixtures
 .vagrant
 .bundle
 vendor
+.ruby-version
+Gemfile.lock
```
Comment (aerostitch) [inline .gitignore]: You should probably do a `git rm Gemfile.lock` to cleanup the repo as well.

File: spec/classes/aerospike_spec.rb (hunk 1 above)
Comment (aerostitch) [inline spec/classes/aerospike_spec.rb]: Nevermind, I got why.

File: .gitignore (hunk 2 above)
Comment (aerostitch) [inline .gitignore:9]: You should probably do a `git rm Gemfile.lock` to cleanup the repo as well.

Comment (deric) [reply .gitignore:9]: It doesn't seem to be versioned, so it wouldn't change anything. `rm Gemfile.lock` is [executed on Travis ](https://github.com/tubemogul/puppet-aerospike/blob/master/.travis.yml#L4) as well, so the test shouldn't be affected.

Comment (aerostitch) [reply .gitignore:9]: My bad, I usually include it in my modules, I didn't see that I didn't put it there this time. :)

**Comment by aerostitch** [comment]
Sounds good. Thanks!

**Pull request merged**